common/include/scipp/common/numeric.h

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2019 Scipp contributors (https://github.com/scipp)
+/// @file
+/// @author Simon Heybrock
+#ifndef SCIPP_COMMON_NUMERIC_H
+#define SCIPP_COMMON_NUMERIC_H
+
+#include <algorithm>
+#include <cmath>
+#include <limits>
+
+#include "scipp/common/index.h"
+
+namespace scipp::numeric {
+
+template <class Range> bool is_linspace(const Range &range) {
+  if (scipp::size(range) < 2)
+    return false;
+  if (range.back() <= range.front())
+    return false;
+
+  using T = typename Range::value_type;
+  const T delta = (range.back() - range.front()) / (scipp::size(range) - 1);
+  constexpr int32_t ulp = 4;
```

Comment:
Would be useful to have the hard-code 4 for Unit of Least Precision explained.

Reply:
Actually I do not have an explanation, I just put something random and small, assuming that "a handful of operations that change by 1 ulp should be allowed". Do you have a guideline how to choose?

Reply:
No I don't have a good suggestion for this choice. Maybe @igudich has a suggestion or method for choosing.